src/core/control/settings/Settings.cpp

```diff
@@ -1331,6 +1348,20 @@ void Settings::setIconTheme(IconTheme iconTheme) {
     save();
 }
 
+auto Settings::getSidebarNumberingStyle() const -> SidebarNumberingStyle { return this->sidebarNumberingStyle; };
+
+void Settings::setSidebarNumberingStyle(SidebarNumberingStyle numberingStyle) {
+    if (this->sidebarNumberingStyle == numberingStyle) {
+        return;
+    }
+
+    this->sidebarNumberingStyle = numberingStyle;
+
+    this->control->getSidebar()->layout();
```

Comment:
This breaks the structure (Settings is essentially a Model class containing all settings, but never acting on anything else).
The solution would rather by to modify `Control::showSettings()`: record the numbering style before calling the SettingsDialog, check once the dialog is done for if that value changed and act accordingly.
The same is done for other parameters in `Control::showSettings()` already. Just add yours.